devito/ir/iet/scheduler.py

```diff
@@ -88,6 +88,7 @@ def _lower_stepping_dims(iet):
             rule = lambda i: i.function.is_TimeFunction and i.function._time_size == k
             replacer = lambda i: xreplace_indices(i, mapper, rule)
             iet = XSubs(replacer=replacer).visit(iet)
+            iet = XSubs_cond(mapper).visit(iet)
```

Comment:
Why at this stage do these 'conditional' substitutions need to be distinguished?